### DISCUSSION TITLE
How to get number of iterations from sparse solvers in JAX

### DISCUSSION THREAD

**Discussion by cagrikymk** [jax-ml/jax#13475] [Q&A] [answered]
Currently I am using the [cg solver](https://jax.readthedocs.io/en/latest/_autosummary/jax.scipy.sparse.linalg.cg.html) in JAX and I am aware that it just returns the solution array and None as info. 
Is there any hacky way to extract the number of iterations till convergence? I use a custom function for SpMV operation for the solve and I would like to know how many times that function is called during the solve. 
I typically do that using callbacks + nonlocal count when using regular scipy when I tried to do something similar in the SpMV function, I just get 5 whether I set max_iter to 1 or 1000. Not sure where that 5 comes from since I can tell that SpMV is called more than 5 times.

It do not mind if it is hacky way since I will use this for only benchmarking. I tried to look at the core JAX library but I failed to find a way to extract such information.

**Comment by PhilipVinc** [comment, marked as answer]
https://gist.github.com/PhilipVinc/34e15d4fc93ee58753c24b29a8fba56e

**Reply by cagrikymk** [reply]
This does exactly what I want!
Thank you
